### PR TITLE
Clean up metric reader options

### PR DESCRIPTION
**Pull request by alanwest**
* Removes `MetricReaderType` enumeration.
* Adds `PeriodicExportingMetricReaderOptions.ExportTimeoutMilliseconds` with a default of `30000` for all push exporters.
* `MetricReaderOptions.PeriodicExportingMetricReaderOptions` is no longer settable.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Console/ConsoleExporterMetricsExtensions.cs [hunk 1]
```diff
@@ -25,6 +26,9 @@ namespace OpenTelemetry.Metrics
     /// </summary>
     public static class ConsoleExporterMetricsExtensions
     {
+        private const int DefaultExportIntervalMilliseconds = Timeout.Infinite;
+        private const int DefaultExportTimeoutMilliseconds = 30000;
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.Console/ConsoleExporterMetricsExtensions.cs:30]: Do we need timeout 30000 or Infinite?

File: src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs [hunk 2]
```diff
@@ -26,6 +27,9 @@ namespace OpenTelemetry.Metrics
     /// </summary>
     public static class InMemoryExporterMetricsExtensions
     {
+        private const int DefaultExportIntervalMilliseconds = Timeout.Infinite;
+        private const int DefaultExportTimeoutMilliseconds = 30000;
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs:31]: Do we need timeout 30000 or Infinite?

**Review by reyang — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]
LGTM except the same qn as Reiley about default for ExportTimeout to be infinite vs 30

**Comment by utpilla** [comment]
We could get rid of the default exportInterval and timeout in `PeriodicExportingMetricReader` now as we would always be providing the parameter values for the `PeriodicExportingMetricReader` ctor:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs#L31-L32

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Exporter.Console/ConsoleExporterMetricsExtensions.cs (hunk 1 above)
Comment (alanwest) [reply src/OpenTelemetry.Exporter.Console/ConsoleExporterMetricsExtensions.cs:30]: I just submitted https://github.com/open-telemetry/opentelemetry-specification/pull/2415 for further clarification.

I'll set to Infinite for now and change depending on how things shake out.

**Comment by alanwest** [comment]
> We could get rid of the default exportInterval and timeout in `PeriodicExportingMetricReader` now as we would always be providing the parameter values for the `PeriodicExportingMetricReader` ctor:

It is still possible to instantiate the `PeriodicExportingMetricReader` by itself, so the defaults are still relevant. Something like

```csharp
var metricExporter = new ConsoleExporter();
var metricReader = new PeriodicExportingMetricReader(exporter);
var meterProviderBuilder = Sdk.CreateMeterProviderBuilder().AddReader(metricReader);
```

**Pull request merged**